src/System.Private.CoreLib/src/System/Reflection/Emit/XXXOnTypeBuilderInstantiation.cs

```diff
@@ -110,10 +110,12 @@ internal override Type[] GetParameterTypes()
             return m_ctor.GetParameterTypes();
         }
 
+#pragma warning disable CS8609 // TODO-NULLABLE: Covariant return types (https://github.com/dotnet/roslyn/issues/23268)
```

Comment:
This should return non-nullable.

You can change the implementation to return `m_type` to avoid going through `DeclaringType`

Reply:
Fixed.